packages/manager/apps/catalog-revamp/src/hooks/useCatalog.ts

```diff
@@ -0,0 +1,83 @@
+import { useState, useEffect } from 'react';
+import { useQuery, UseQueryResult } from '@tanstack/react-query';
+import {
+  getManagerHubCatalogList,
+  getManagerHubCatalogListQueryKey,
+} from '@/api';
+import {
+  groupItems,
+  filterByEnum,
+  getAvailableCategories,
+  matchCategories,
+  matchUniverses,
+  matchSearchText,
+  filterItems,
+  getUniverses,
+} from '@/utils/utils';
+
+export type Product = {
+  id: number;
+  name: string;
+  description: string;
+  lang: string;
+  categories: string[];
+  url: string;
+  regionTags: string[];
+  productName: string;
+  order: string;
+  universe: string;
+  category: string;
+};
+
+interface ServiceData {
+  status: number;
+  data: any;
+}
+
+interface CatalogProps {
+  selectedCategories: string[];
+  selectedUniverses: string[];
+  searchText: string;
+}
+
+export const useCatalog = ({
+  selectedCategories,
+  selectedUniverses,
+  searchText,
+}: CatalogProps) => {
+  const [products, setProducts] = useState([]);
+  const [items, setItems] = useState(<any>[]);
+  const [categories, setCategories] = useState([]);
+  const [universes, setUniverses] = useState([]);
```

Comment:
you can delete it and manage it in the parent component

Reply:
Done